src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -576,7 +576,7 @@ public BlockHeader FindHeader(long number, BlockTreeLookupOptions options)
 
         public Keccak FindBlockHash(long blockNumber) => GetBlockHashOnMainOrBestDifficultyHash(blockNumber);
 
-        public BlockHeader FindHeader(Keccak blockHash, BlockTreeLookupOptions options)
+        public BlockHeader? FindHeader(Keccak? blockHash, BlockTreeLookupOptions options)
```

Comment:
These both were always nullable, but not flagged as such.  Nethermind.Blockchain project didn't have null checking enabled and when I enabled it (see below) these popped up so I corrected the code so it is at least clear what is happening.  Interestingly, the implementation for one of these has a comment about "why is null still possible here?" which could be tracked down via C# static analysis of nulls, which first requires enabling it and fixing all of the things like this.

Reply:
Nullability is a bigger cleanup task -> initially Nethermind was not written with nullables because they simply did not exist yet. I started marking more an more proejcts with nullability checks until one day I cause a serious trouble by changing entire state management code to nullables. After that bad experience I stopped touch more critical parts of the system and left it for quieter days. Much work has been done, much is left. There are probably around 10000 - 100000 minor changes to the codebase to make it fully check nullables. It is a kind of work that I actually really like spending time on but I believe it is around 30 hours to do it all.